### PR TITLE
Add oblique mercator to `create_bounds` and other improvements to the regridding function

### PR DESCRIPTION
**Pull request by RondeauG**
### Pull Request Checklist:
- [x] This PR addresses an already opened issue (for bug fixes / features)
    - This PR fixes #xyz
- [x] (If applicable) Documentation has been added / updated (for bug fixes / features).
- [x] (If applicable) Tests have been added.
- [x] This PR does not seem to break the templates.
- [x] CHANGELOG.rst has been updated (with summary of main changes).
  - [x] Link to issue (:issue:`number`) and pull request (:pull:`number`) has been added.

### What kind of change does this PR introduce?

* Renames `create_bounds_rotated_pole` to `create_bounds_gridmapping` and adds support for `oblique_mercator`.
* Changes to `regrid_dataset`:
  * `weights_location` is now optional.
  * Reinforced the automatic naming of the weights file. When attributes are unknown, a random string is now generated instead of a static one. Moreover, it now relies on a `cat:id` and `cat:domain` for both the incoming dataset and the grid.
  * Now correctly skips the `lon_bounds` and `lat_bounds` variables if the grid was generated using `xesmf.util.cf_grid_2d`
* `create_mask` has been completely refactored to use real arguments instead of a dictionary.

### Does this PR introduce a breaking change?

* The old function name is deprecated and will be dropped in a later version.
* `weights_location` in `regrid_dataset` is now optional, and thus had to be moved to explicit arguments rather than positional.

### CODE REVIEW

File: environment-dev.yml [hunk 1]
```diff
@@ -29,8 +29,8 @@ dependencies:
   - shapely >=2.0
   - sparse
   - toolz
-  - xarray >=2023.11.0, !=2024.6.0
-  - xclim >=0.52.2, <0.53
+  - xarray >=2023.11.0,!=2024.6.0
+  - xclim >=0.52.2,<0.53
```
Comment (RondeauG) [inline environment-dev.yml:33]:
```suggestion
  - xarray >=2023.11.0, !=2024.6.0
  - xclim >=0.52.2, <0.53
```

File: environment.yml [hunk 2]
```diff
@@ -29,8 +29,8 @@ dependencies:
   - shapely >=2.0
   - sparse
   - toolz
-  - xarray >=2023.11.0, !=2024.6.0
-  - xclim >=0.52.2, <0.53
+  - xarray >=2023.11.0,!=2024.6.0
+  - xclim >=0.52.2,<0.53
```
Comment (RondeauG) [inline environment.yml:33]:
```suggestion
  - xarray >=2023.11.0, !=2024.6.0
  - xclim >=0.52.2, <0.53
```

File: pyproject.toml [hunk 3]
```diff
@@ -64,8 +64,8 @@ dependencies = [
   "shapely >=2.0",
   "sparse",
   "toolz",
-  "xarray >=2023.11.0, !=2024.6.0",
-  "xclim >=0.52.2, <0.53",
+  "xarray >=2023.11.0,!=2024.6.0",
+  "xclim >=0.52.2,<0.53",
```
Comment (RondeauG) [inline pyproject.toml:68]:
```suggestion
  "xarray >=2023.11.0, !=2024.6.0",
  "xclim >=0.52.2, <0.53",
```

File: src/xscen/regrid.py [hunk 4]
```diff
@@ -350,54 +372,104 @@ def _regridder(
...
+            pole_longitude=float(ds.rotated_pole.grid_north_pole_longitude),
+            pole_latitude=float(ds.rotated_pole.grid_north_pole_latitude),
+            central_rotated_longitude=central,
+        )
+    elif gridmap == "oblique_mercator":
+        RP = ccrs.ObliqueMercator(
+            central_longitude=float(ds.oblique_mercator.longitude_of_projection_origin),
+            central_latitude=float(ds.oblique_mercator.latitude_of_projection_origin),
+            false_easting=float(ds.oblique_mercator.false_easting),
+            false_northing=float(ds.oblique_mercator.false_northing),
```
Comment (aulemahal) [inline src/xscen/regrid.py:420]: Ces deux-là sont marqués comme optionnel dans la doc de CF, ce serait bien de faire comme en haut? Ou en fait:
```suggestion
            false_easting=float(ds.oblique_mercator.attrs.get('false_easting', 0)),
            false_northing=float(ds.oblique_mercator.attrs.get('false_northing', 0)),
```

**Review by aulemahal — APPROVED** [review]
This looks good!

File: src/xscen/regrid.py [hunk 5]
```diff
@@ -350,54 +372,104 @@ def _regridder(
...
     )
-    if not Path(filename).is_file():
+    if filename is not None and not Path(filename).is_file():
         regridder.to_netcdf(filename)
 
     return regridder
 
 
-def create_bounds_rotated_pole(ds: xr.Dataset):
+def create_bounds_rotated_pole(ds: xr.Dataset) -> xr.Dataset:
```
Comment (juliettelavoie) [inline src/xscen/regrid.py:381]: on ne le voit pas sur RtD (ce n'est pas nouveau de cette PR, mais j'ai remarquer ça cette semaine). Peut-être parce que c'est sous une fontion "_" ?

File: src/xscen/regrid.py [hunk 6]
```diff
@@ -350,54 +372,104 @@ def _regridder(
...
+    warnings.warn(
+        "This function is deprecated and will be removed in xscen v0.12.0. Use create_bounds_gridmapping instead.",
+        FutureWarning,
+    )
+    return create_bounds_gridmapping(ds, "rotated_pole")
+
+
+def create_bounds_gridmapping(ds: xr.Dataset, gridmap: str) -> xr.Dataset:
```
Comment (juliettelavoie) [inline src/xscen/regrid.py:389]: idem

**Review by juliettelavoie — APPROVED** [review]

File: src/xscen/regrid.py [hunk 7]
```diff
@@ -21,20 +24,15 @@
 
 from .config import parse_config
 
-# TODO: Implement logging, warnings, etc.
-# TODO: Add an option to call xesmf.util.grid_2d or xesmf.util.grid_global
-# TODO: Implement support for an OBS2SIM kind of interpolation
-
-
 __all__ = ["create_mask", "regrid_dataset"]
```
Comment (aulemahal) [inline src/xscen/regrid.py:27]:
```suggestion
__all__ = ["create_bounds_gridmapping", "create_mask", "regrid_dataset"]
```

File: src/xscen/regrid.py (hunk 4 above)
Comment (RondeauG) [reply src/xscen/regrid.py]: Je ne voulais pas mettre de valeur juste au cas où le défaut change plus tard. J'ai créé une petite sous-fonction pour que ça soit plus clean.

**Pull request merged**